flink/src/main/java/org/apache/iceberg/flink/source/DataIterator.java

```diff
@@ -21,64 +21,36 @@
 
 import java.io.IOException;
 import java.io.UncheckedIOException;
-import java.nio.ByteBuffer;
 import java.util.Iterator;
-import java.util.Map;
-import java.util.stream.Stream;
 import org.apache.iceberg.CombinedScanTask;
 import org.apache.iceberg.FileScanTask;
-import org.apache.iceberg.encryption.EncryptedFiles;
-import org.apache.iceberg.encryption.EncryptedInputFile;
 import org.apache.iceberg.encryption.EncryptionManager;
+import org.apache.iceberg.encryption.InputFilesDecryptor;
 import org.apache.iceberg.io.CloseableIterator;
 import org.apache.iceberg.io.FileIO;
-import org.apache.iceberg.io.InputFile;
-import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
-import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
-import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 
 /**
  * Base class of Flink iterators.
  *
  * @param <T> is the Java class returned by this iterator whose objects contain one or more rows.
  */
-abstract class DataIterator<T> implements CloseableIterator<T> {
+public class DataIterator<T> implements CloseableIterator<T> {
```

Comment:
Why make this internal class  public ?

Reply:
This is made public because flip-27 source code will need to construct it from a diff package like `org.apache.iceberg.flink.source.reader`.  I can add a `@Internal` annotation to those classes.